src/libraries/Microsoft.PowerFx.Interpreter/ReflectionFunction.cs

```diff
@@ -127,6 +127,40 @@ public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationTok
         }
     }
 
+    internal class CustomAsyncTexlFunction : TexlFunction, IAsyncTexlFunction
+    {
+        public Func<FormulaValue[], CancellationToken, Task<FormulaValue>> _impl;
+
+        public override bool SupportsParamCoercion => true;
+
+        public CustomAsyncTexlFunction(string name, FormulaType returnType, params FormulaType[] paramTypes)
+            : this(name, returnType._type, Array.ConvertAll(paramTypes, x => x._type))
+        {
+        }
+
+        public CustomAsyncTexlFunction(string name, DType returnType, params DType[] paramTypes)
+            : base(DPath.Root, name, name, SG("Custom func " + name), FunctionCategories.MathAndStat, returnType, 0, paramTypes.Length, paramTypes.Length, paramTypes)
+        {
+        }
+
+        public override bool IsSelfContained => true;
```

Comment:
Are these necessarily always side effect-free, or should we ask implementers to provide this in the constructor?